### PR TITLE
devicestate: log checkEncryption errors via logger.Noticef

### PR DESCRIPTION
**Pull request by mvo5**
This is a followup for PR#9791 that logs errors from
checkEncryption to the logger during install.

### CODE REVIEW

File: overlord/devicestate/handlers_install.go [hunk 1]
```diff
@@ -350,6 +350,9 @@ func (m *DeviceManager) checkEncryption(st *state.State, deviceCtx snapstate.Dev
 			return false, fmt.Errorf("cannot encrypt device storage as mandated by encrypted storage-safety model option: %v", checkEncryptionErr)
 		}
 
+		// log any errors even if they are not fatal
+		logger.Noticef("ignoring error from check encryption: %v", checkEncryptionErr)
```
Comment (pedronis) [inline overlord/devicestate/handlers_install.go:354]: maybe we want two different logs depending on hasFDESetupHook:

not encrypting device storage as querying kernel fde-setup hook did not succeed: %v

not encrypting device storage as checking TPM gave: %v

perhaps?

Comment (anonymouse64) [reply overlord/devicestate/handlers_install.go:354]: agreed I think that would be nicer

Comment (mvo5) [reply overlord/devicestate/handlers_install.go]: Thanks, I updated the code for this now.

**Review by pedronis — APPROVED** [review]
thank you

**Pull request merged**